### PR TITLE
A better fix for Google Play Services taking project properties

### PR DESCRIPTION
**Pull request by yangtaufoo**
An better extension of #858!

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 
+def DEFAULT_COMPILE_SDK_VERSION             = 26
+def DEFAULT_BUILD_TOOLS_VERSION             = "26.0.1"
+def DEFAULT_TARGET_SDK_VERSION              = 26
+def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION    = "18.+"
+
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.2"
+    compileSdkVersion project.hasProperty('compileSdkVersion') ? project.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
```
Comment (evollu) [inline android/build.gradle]: I'm not familiar with gradle settings. curious here
1. is project.hasProperty('compileSdkVersion') reading from main project so it is always defined?
2. what is "18.+" for? I thought the latest is 12?

**Comment by yangtaufoo** [comment]
project.hasProperty(*) is reading the optional variables from the file android/grade.properties. 

As it is optional, the user might not define it. As a fail safe when users do not define the variables, the default variables are there to make sure that compilation works.

I’ve made an error with the Google Play Services version. It should be 11.+ instead of 18.+

File: android/build.gradle [hunk 2]
```diff
@@ -1,12 +1,17 @@
 apply plugin: 'com.android.library'
 
+def DEFAULT_COMPILE_SDK_VERSION             = 26
```
Comment (evollu) [inline android/build.gradle]: can you change it back to 25? inline with react-native default

Comment (yangtaufoo) [reply android/build.gradle]: Alright, I will do it. Any other requests?

Comment (evollu) [reply android/build.gradle]: that's it

Comment (yangtaufoo) [reply android/build.gradle]: Done, thanks @evollu for reviewing in such short notice. Might want to update the readme to have a section where users can define their own Google Play Services version as per outlined in Android documentation (https://developer.android.com/studio/build/gradle-tips.html#configure-project-wide-properties)

**Pull request merged**